src/en/clowncorps/src/eu/kanade/tachiyomi/extension/en/clowncorps/ClownCorps.kt

```diff
@@ -0,0 +1,175 @@
+    override val lang = "en"
+    override val name = "Clown Corps"
+    override val supportsLatest = false
+
+    private val creator = "Joe Chouinard"
+
+    // Text from: https://clowncorps.net/about/
+    private val synopsis = "Clown Corps is a comic about crime-fighting clowns. " +
+        "It's pronounced \"core.\" Like marine corps."
+
+    override val client = network.client.newBuilder()
+        .addInterceptor(TextInterceptor())
+        .build()
+
+    override fun fetchPopularManga(page: Int): Observable<MangasPage> = Observable.fromCallable {
+        MangasPage(
+            listOf(
+                SManga.create().apply {
+                    title = name
+                    artist = creator
+                    author = creator
+                    description = synopsis
+                    status = SManga.ONGOING
+                    // Image from: https://clowncorps.net/about/
+                    thumbnail_url = "$baseUrl/wp-content/uploads/2022/11/clowns41.jpg"
+                    setUrlWithoutDomain("/comic")
+                },
+            ),
+            hasNextPage = false,
+        )
+    }
+
+    override fun fetchSearchManga(page: Int, query: String, filters: FilterList) =
+        fetchPopularManga(page)
+
+    override fun fetchMangaDetails(manga: SManga): Observable<SManga> =
+        Observable.just(manga.apply { initialized = true })
+
+    override fun chapterListParse(response: Response): List<SChapter> {
+        val pageStatus = response.asJsoup().select("#paginav li.paginav-pages").text()
+        val pageCount = pageStatus.split(" ").last().toInt()
+
+        val chapters = mutableListOf<SChapter>()
+
+        for (page in 1..pageCount) {
+            val url = "$baseUrl/comic/page/$page/"
+            val resp = client.newCall(GET(url, headers)).execute()
+            val comics = resp.asJsoup().select(".comic")
+            for (comic in comics) {
+                val link = comic.selectFirst(".post-title a")?.attr("href")
+                val title = comic.selectFirst(".post-title a")?.text()
+                val postDate = comic.selectFirst(".post-date")?.text()
+                val postTime = comic.selectFirst(".post-time")?.text()
+                if (link == null || title == null || postDate == null || postTime == null) continue
+                val date = parseDate("$postDate $postTime")
+
+                val chapter = SChapter.create().apply {
+                    setUrlWithoutDomain(link)
+                    name = title
+                    date_upload = date
+                }
+                chapters.add(chapter)
+            }
+        }
+
+        return chapters
+    }
+
+    private fun parseDate(dateStr: String): Long {
+        return try {
+            dateFormat.parse(dateStr)!!.time
+        } catch (_: ParseException) {
+            0L
+        }
+    }
+
+    private val dateFormat by lazy {
+        SimpleDateFormat("MMMM dd, yyyy hh:mm aa", Locale.ENGLISH)
+    }
```

Comment:
```suggestion
    override fun chapterListParse(response: Response): List<SChapter> {
        return response.asJsoup().select("#chapter-select .level-0").flatMap { chapterElement ->
            val url = chapterElement.attr("value")
            val document = client.newCall(GET(url, headers)).execute().asJsoup()
            document.select("#ceo_comic_list_dropdown_widget-6 .level-0").map {
                SChapter.create().apply {
                    setUrlWithoutDomain(it.attr("value"))
                    name = it.text()
                }
            }
        }.asReversed()
    }
```

Reply:
This does not set the date. Don't we want that?

Reply:
Your implementation is too slow - it will iterate through the 45 pages and send that many requests. That puts a lot of stress onto the source server.

Reply:
That's true and fair.
Is there nothing else that could be done for this, though?

Reply:
Oh, I just found something new!
Maybe we could loop over the years?
https://clowncorps.net/2024/?post_type=comic

Reply:
Nothing much we can do. The current extension API doesn't provide the existing chapter list, so we can only fetch from scratch.

You can hack it by storing chapter info into shared preferences (considering you have only one manga entry) but code logic to do patch updates will be complicated and error-prone.

Reply:
> Oh, I just found something new!\nMaybe we could loop over the years?\nhttps://clowncorps.net/2024/?post_type=comic

That's still 10 pages per webpage.

Reply:
Drat

Reply:
I added caching in 6ed7adf6be7170a20bdb913e9d115bbd18d67542, so hopefully that should lessen the load on the server enough to the point that my original method can still be used.

Quick summary of the way this caching works:
The first time, it loads all (45 as of the time of writing) webpages, and collects the chapters from those.
It then saves all of those chapters in the SharedPreferences, serialized to JSON.

The next time the comic gets refreshed, it gets all the chapters it can from the cache, and stores them in a **set**.
It then fetches every webpage starting at 1 again, which is where the most recent new chapters get added to.
Then, it continues going through every next page (thus going further and further into the past).
It keeps adding the newly fetched chapters into the set, but when all the chapters it just fetched are already in the set, it knows it has reached where the cache left off, so it stops the refresh.
If you regularly update, that means it only ever does one request.

Hopefully this explanation makes the code make a bit more sense.

I also added a button in the extension preferences menu to clear the cache.